src/raw_node.rs

```diff
@@ -257,6 +257,11 @@ impl<T: Storage> RawNode<T> {
 
     // Propose proposes data be appended to the raft log.
     pub fn propose(&mut self, data: Vec<u8>, sync_log: bool) -> Result<()> {
+        self.propose_with_context(data, sync_log, 0)
```

Comment:
I suggest to remove `sync_log` from the `propose` signature, since it's part of the context, which should be handled by users.